File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<int> judge(n + 1, 1);
        vector<int> indeg(n + 1, 0);
        for(vector<int> v: trust)
        {
            int a = v[0], b = v[1];
            indeg[b]++;
            judge[a] = 0;
        }
        for(int i = 1; i <= n; i++)
        {
            if(judge[i] == 1 && indeg[i] == n - 1) return i;
        }
        return -1;
    }
};

File: 0997-find-the-town-judge/0997-find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        notJudge = set()
        cand = 1
        for a, b in trust:
            notJudge.add(a)
            if b not in notJudge:
                cand = b
        if cand in notJudge: return -1
        return cand if all([[i, cand] in trust for i in range(1, n+1) if i != cand]) else -1

